pkg/controller/controller.go

```
@@ -95,6 +117,29 @@ func NoopFunc(ctx context.Context) error {
 	return nil
 }
 
+// isGroupMetricEnabled returns true if metrics are enabled for the Group
+//
+// The controller metrics config option is used to determine
+// if "all", "none" (takes precedence over "all"), or the
+// given set of Group names should be enabled.
+//
+// If no controller metrics config option was provided,
+// only then is the DefaultMetricsEnabled field used.
+func isGroupMetricEnabled(g Group) bool {
+	var metricsEnabled = groupMetricEnabled
+	if metricsEnabled[noControllerMetricsEnabled] {
+		// "none" takes precedence over "all"
+		return false
+	} else if metricsEnabled[allControllerMetricsEnabled] {
+		return true
+	} else if len(metricsEnabled) > 0 {
+		// if any configuration was provided,
+		// then do not use any of the defaults
+		return metricsEnabled[g.Name]
+	}
+	return false
```

I think there's no need to check the map length. Also, all the `else` can be removed, since we are early-returning:
```
	if metricsEnabled[noControllerMetricsEnabled] {
		// "none" takes precedence over "all"
		return false
	}
	if metricsEnabled[allControllerMetricsEnabled] {
		return true
	}
	// if any configuration was provided,
	// then do not use any of the defaults
	return metricsEnabled[g.Name]
```
Besides, the last comment about the defaults is unclear to me. By "defaults" you mean "all" and "none"? If so, I think the comment is misleading, since we are looking for them first, in the map.

---

Thanks for the pointer. There were previously some hardcoded defaults, so I can see how the comments are confusing.